utils/lib/Alert.ts

```diff
@@ -81,6 +140,92 @@ class Alert extends Component<QuickstartConfigAlert[], QuickstartAlertInput[]> {
       };
     });
   }
+
+  /**
+   * Static method that gets the alert policy associated with a quickstart and it's current data sources
+   * @returns - object with alert policy ids, required data sources and NG errors
+   */
+  static async getAlertPolicyRequiredDataSources(quickstart: {
+    name: string;
+    dataSourceIds: string[];
+  }): Promise<
+    | { alertPolicy: AlertPolicyDataSource }
+    | { alertPolicy: null; errors: ErrorOrNerdGraphError[] }
+  > {
+    const { data, errors } = await fetchNRGraphqlResults<
+      AlertPolicyRequiredDataSourcesQueryVariables,
+      AlertPolicyRequiredDataSourcesQueryResults
+    >({
+      queryString: ALERT_POLICY_REQUIRED_DATA_SOURCES_QUERY,
+      variables: { query: `${quickstart.name} alert policy` },
+    });
+
+    const results = data?.actor?.nr1Catalog?.search?.results;
+
+    if (errors) {
+      return { alertPolicy: null, errors };
+    }
+
+    if (results === undefined || results.length === 0) {
+      const error = new Error(
+        `No alert policy for quickstart ${quickstart.name} exists`
+      );
+
+      return { alertPolicy: null, errors: [error] };
+    }
+
+    const hasFailed = quickstart.dataSourceIds.length > 1;
```

Comment:
[part 1] actually thinking about this more I feel like since `quickstart` is an input of the function, we should actually return hasFailed earlier so we don't make the request if there are multiple data sources detected for a quickstart, where i think we can put this I commented in `utils/set-alert-policy-required-datasources.ts` as part 2

Reply:
The reason we do it within the get alert policy function is because we need to know which alert policy is associated with that quickstart so that we know which alert policy to manually update. So I'm not sure we should move it to before we get the alert policy.

Reply:
oooooooooo gotcha yes that makes sense